lib/serializer/cbor/aws_iot_serializer_tinycbor_encoder.c

```diff
@@ -145,19 +177,23 @@ static AwsIotSerializerError_t _openContainer( AwsIotSerializerEncoderObject_t *
                                                AwsIotSerializerEncoderObject_t * pNewEncoderObject,
                                                size_t length )
 {
-    AwsIotSerializerError_t returnError = AWS_IOT_SERIALIZER_INTERNAL_FAILURE;
-    CborError cborError = CborUnknownError;
+    /* New object must be a container. */
+    if( !AwsIotSerializer_IsContainer( pNewEncoderObject->type ) )
+    {
+        return AWS_IOT_SERIALIZER_INVALID_INPUT;
+    }
 
-    CborEncoder * pOuterEncoder = _castEncoderObjectToCborEncoder( pEncoderObject );
+    AwsIotSerializerError_t returnedError = AWS_IOT_SERIALIZER_SUCCESS;
```

Comment:
Declarations are at the beginning of the function as per the coding standard.

Reply:
I wonder if this standard can be loosen a little bit. I found it more readable in some cases...